src/Stratis.Bitcoin/P2P/Peer/NetworkPeer.cs

```diff
@@ -530,10 +530,12 @@ private async Task ProcessMessageAsync(IncomingMessage message, CancellationToke
                 return;
             }
 
-            try
-            {
+            bool insideCallback = this.onDisconnectedAsyncContext.Value != null;
```

Comment:
I believe processes message is always processed by a single thread (sequentially) and will never be called from one of the behaviours so `this.onDisconnectedAsyncContext.Value` will never have a value (if I am not mistaken)

Reply:
Perhaps we could put this PR on ice and instead create a PR that adds a `Guard.Assert` to ensure that the value is always null.